### PR TITLE
Blog post - New extension: Hibernate Search + Elasticsearch

### PR DESCRIPTION
**Pull request by gsmet**

### CODE REVIEW

File: _posts/2019-06-05-hibernate-search-extension.adoc [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+tags: quarkus hibernate elasticsearch
+author: gsmet
+---
+
+In the 0.15.0 version of Quarkus, we introduced the Hibernate Search + Elasticsearch extension and, just a few days ago, we added link:/guides/hibernate-search-guide[a guide] and https://github.com/quarkusio/quarkus-quickstarts/tree/master/hibernate-search-elasticsearch[a quickstart] explaining how to use it.
+
+Note that this extension is still experimental as it is based on an Alpha version of Hibernate Search 6.
+While the code is stable and thoroughly tested, some features are still missing and we might change some APIs along the way to the final version - even if the Hibernate team has made a great effort to stabilize them before the inclusion in Quarkus.
+
+If you don't mind updating your code from time to time, you're very welcome to become an early adopter and provide feedback.
```
Comment (rsvoboda) [inline _posts/2019-06-05-hibernate-search-extension.adoc]: What are the benefits of Hibernate Search 6 ? 
In short better marketing why people should jump on this ...

This line mentions drawback of updating your code from time to time and expectation to provide feedback.

Comment (gsmet) [reply _posts/2019-06-05-hibernate-search-extension.adoc]: Well, it's basically Search redesigned to support both Lucene and Elasticsearch properly. Right now, apart from the fact that it's very different, you won't see a lot of advantages. They will come later.

Comment (rsvoboda) [reply _posts/2019-06-05-hibernate-search-extension.adoc]: If you don't mind updating your code from time to time, you're very welcome to become an early adopter and provide feedback. 
  =>
If you want proper support for both Lucene and Elasticsearch and don't mind updating your code from time to time, you're very welcome to become an early adopter and provide feedback.

wdyt ?

File: _posts/2019-06-05-hibernate-search-extension.adoc [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+public class Book extends PanacheEntity {
+
+    @FullTextField(analyzer = "standard") // <2>
+    public String title;
+}
+----
+<1> Declare your entity as part of the full text index
+<2> Add a full text field using the `standard` analyzer provided by Elasticsearch
+
+Then you need some additional configuration:
```
Comment (rsvoboda) [inline _posts/2019-06-05-hibernate-search-extension.adoc]: I would point to `src/main/resources/application.properties`

New Quarkus users may not be aware of that file

File: _posts/2019-06-05-hibernate-search-extension.adoc (hunk 1 above)
Comment (rsvoboda) [reply _posts/2019-06-05-hibernate-search-extension.adoc]: ah, finished reading the blog post, the above proposed change doesn't make sense, new proposal:

If you want proper support for Elasticsearch on Quarkus and don't mind updating your code from time to time, you're very welcome to become an early adopter and provide feedback.

File: _posts/2019-06-05-hibernate-search-extension.adoc [hunk 3]
```diff
@@ -0,0 +1,124 @@
+---
+layout: post
+title: 'New extension: Hibernate Search + Elasticsearch'
+date: 2019-06-05
+tags: quarkus hibernate elasticsearch
+author: gsmet
+---
+
+In the 0.15.0 version of Quarkus, we introduced the Hibernate Search + Elasticsearch extension and, just a few days ago, we added link:/guides/hibernate-search-guide[a guide] and https://github.com/quarkusio/quarkus-quickstarts/tree/master/hibernate-search-elasticsearch[a quickstart] explaining how to use it.
```
Comment (emmanuelbernard) [inline _posts/2019-06-05-hibernate-search-extension.adoc]: I'd recommend you don't do just a few days ago. Blogs are for ever (ish).

File: _posts/2019-06-05-hibernate-search-extension.adoc [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+Note that this extension is still experimental as it is based on an Alpha version of Hibernate Search 6.
```
Comment (emmanuelbernard) [inline _posts/2019-06-05-hibernate-search-extension.adoc]: s/Note that t/T/

File: _posts/2019-06-05-hibernate-search-extension.adoc [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+While the code is stable and thoroughly tested, some features are still missing and we might change some APIs along the way to the final version - even if the Hibernate team has made a great effort to stabilize them before the inclusion in Quarkus.
```
Comment (emmanuelbernard) [inline _posts/2019-06-05-hibernate-search-extension.adoc]: I'd put the whole experimental aspect down toe blog.
Or consider that Quarkus is a 0.x and everything is experimental to some extends. 
Come to think of it, I'd make sure people want to try in the blog and have the guide be the cautious provider.

File: _posts/2019-06-05-hibernate-search-extension.adoc [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+It can't be easier.
+
+First, add the `quarkus-hibernate-search-elasticsearch` extension to your Quarkus application:
+
+[source, shell]
+----
+./mvnw quarkus:add-extension -Dextension=hibernate-search-elasticsearch
+----
+
+(or you can directly add the `io.quarkus:quarkus-hibernate-search-elasticsearch` dependency to your `pom.xml`)
```
Comment (emmanuelbernard) [inline _posts/2019-06-05-hibernate-search-extension.adoc]: you can leave it but remember this is hyper opinionated so people can learn the alternatives in guides

File: _posts/2019-06-05-hibernate-search-extension.adoc (hunk 2 above)
Comment (emmanuelbernard) [reply _posts/2019-06-05-hibernate-search-extension.adoc]: +1 for pointing the file name.

Also try and ban "You need" form your vocabulary. Just say it plain. "Then add some configuration"

Need, must, highly experimental and going to make the reptilian brain go in fly mode. Not what you want in a blog.

File: _posts/2019-06-05-hibernate-search-extension.adoc [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+== Going further
+
+This is just a small taste of what you can do with our Hibernate Search + Elasticsearch extension.
+
+For instance, understanding the concept of text analysis is especially important for full text search and properly setting up analyzers depending on your requirements necessitates a bit more work.
+
+You will find a lot more information about how you can set everything up properly in link:/guides/hibernate-search-guide[our extensive guide].
+
+== FAQ
```
Comment (emmanuelbernard) [inline _posts/2019-06-05-hibernate-search-extension.adoc]: I'd point to the guide and move the FAQ there if you think it's important to have.

**Comment by emmanuelbernard** [comment]
It's good.
I added comments to make gooder ;)

I think you could add a section on how it's more awesome in Quarkus than elsewhere if the extension does some build time optimisations like reading the metadata etc. That would tie it to Quarkus being a great platform to run Hibernate Search on.

File: _posts/2019-06-05-hibernate-search-extension.adoc (hunk 6 above)
Comment (gsmet) [reply _posts/2019-06-05-hibernate-search-extension.adoc]: Yeah, I left it because using the Maven command reformats your pom file.

**Pull request merged**